plugins/hypervisors/xenserver/src/com/cloud/hypervisor/xenserver/resource/CitrixResourceBase.java

```diff
@@ -3332,7 +3332,14 @@ public PowerState getVmState(final Connection conn, final String vmName) {
                     vmStatsAnswer.setDiskReadKBs(vmStatsAnswer.getDiskReadKBs() + getDataAverage(dataNode, col, numRows) / 1000);
                 } else if (param.matches("vbd_.*_write")) {
                     vmStatsAnswer.setDiskWriteKBs(vmStatsAnswer.getDiskWriteKBs() + getDataAverage(dataNode, col, numRows) / 1000);
+                } else if (param.contains("memory_internal_free")) {
```

Comment:
I know you just followed the way the code was written here before, but this “if/else/if/else……” structure is too big; I believe to introduce some changes like this here; it would be much appreciated a refactoring. I do not mean a huge refactoring, only the extraction of this if/else structure to a method and then using some other clever way to do these checks and to retrieve the values.